### DISCUSSION TITLE
How to increase HEIF max loadable resolution?

### DISCUSSION THREAD

**Discussion by FluffyDiscord** [libvips/libvips#3635] [Q&A] [answered]
Hi, I have just noticed that my conversion pipeline started failing some time ago. I am getting error `bad seek to 436511 heif: Memory allocation error: Security limit exceeded: Image size 2625x19832 exceeds the maximum image size 16384x16384` (mind that this is produced by php bindings https://github.com/libvips/php-vips)

If I understand it correctly, I need to increase the max size with `heif_context_set_maximum_image_size_limit` ? Would someone please tell me where/how I can increase it? I am using Docker containers, if that helps in any way.

**Comment by jcupitt** [comment, marked as answer]
Hi, just load with the `unlimited` flag set, eg.:

```
$image = Vips\Image::heifload("x.heic", ["unlimited" => true]);
```

**Reply by FluffyDiscord** [reply]
That should do. Thank you for your time

**Reply by FluffyDiscord** [reply]
Would it be possible to add this option/flag to `Vips\Image::thumbnail()` too ?

**Reply by jcupitt** [reply]
You can add it to the filename, eg.:

```php
$thumb = Vips\Image::thumbnail($filename . "[unlimited]", 256);
```
